service/controller/chart/v1/resource/release/resource.go

```diff
@@ -112,14 +112,22 @@ func (r *Resource) patchAnnotations(ctx context.Context, cr v1alpha1.Chart, rele
 	currentChecksum := key.ValuesMD5ChecksumAnnotation(*currentCR)
 
 	if releaseState.ValuesMD5Checksum != currentChecksum {
-		patches := []Patch{
-			{
+		patches := []Patch{}
+
```

Comment:
If the chart CR has no annotations we need to add the empty map. As nested keys are not created.